src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,58 @@
+package duke.tasks;
+
+import duke.exceptions.IncompleteDescriptionException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * A class encapsulating information of a task of Deadline type.
+ */
+public class Deadline extends Task {
+
+    private String deadline;
+    private LocalDate ld;
+
+    public Deadline(String taskName, String deadline) throws IncompleteDescriptionException {
+        super(taskName);
+        try {
+            this.ld = LocalDate.parse(deadline);
+        } catch (DateTimeParseException ignored) {}
+        this.deadline = deadline;
+    }
+
+    /**
+     * Factory method for the deadline class.
+     *
+     * @param taskDesc Description of the deadline task.
+     * @return A Deadline task object.
+     * @throws IncompleteDescriptionException If the task description or task name is empty.
+     */
+    public static Deadline create(String taskDesc) throws IncompleteDescriptionException {
+        String[] tmp = taskDesc.split(" /by ");
+        if (tmp.length <= 1) {
```

Comment:
Perhaps it would be clearer to separate logical units within a block with one blank line!